crates/bevy_pbr/src/light.rs

```diff
@@ -606,6 +607,21 @@ pub struct NotShadowCaster;
 #[reflect(Component, Default)]
 pub struct NotShadowReceiver;
 
+/// Add this component to a [`Camera3d`](bevy_core_pipeline::core_3d::Camera3d)
+/// to control how to smooth the edges of shadows.
+#[derive(Component, ExtractComponent, Reflect, Clone, Copy, PartialEq, Eq, Default)]
+#[reflect(Component, Default)]
+pub enum ShadowSmoothingMode {
+    /// No smoothing (fastest, but jagged edges).
+    NoSmoothing,
+    /// Smooth shadows using the default algorithm.
+    #[default]
+    Smooth,
+    /// Smooth shadows meant to be used in conjunction with
+    /// [`TemporalAntiAliasSettings`](bevy_core_pipeline::experimental::taa::TemporalAntiAliasSettings).
+    Stochastic,
+}
```

Comment:
The general method is called percentage closer filtering, I think we should mention that. Also, I haven't seen it called smoothing, only filtering. I think being consistent in terminology is important so that people can leverage transferrable knowledge about rendering from other contexts than bevy. Please update the names everywhere. :)

```suggestion
/// Add this component to a [`Camera3d`](bevy_core_pipeline::core_3d::Camera3d)
/// to control how to filter shadows to remove aliasing from their edges. The different modes use different
/// approaches to [Percentage Closer Filtering](https://developer.nvidia.com/gpugems/gpugems/part-ii-lighting-and-shadows/chapter-11-shadow-map-antialiasing)
#[derive(Component, ExtractComponent, Reflect, Clone, Copy, PartialEq, Eq, Default)]
#[reflect(Component, Default)]
pub enum ShadowFilteringMethod {
    /// Hardware 2x2 PCF (fastest, but jagged edges).
    Hardware2x2,
    /// The filtering method implemented in The Witness which produces soft edges.
    #[default]
    TheWitness,
    /// A statistical filtering method that appears noisy, but works well when used with
    /// [`TemporalAntiAliasSettings`](bevy_core_pipeline::experimental::taa::TemporalAntiAliasSettings).
    Stochastic,
}
```

Reply:
I'm not sure I like the proposed changes. It's more technical, but less friendly to non-graphics people. Using technical names for features was always my biggest gripe with existing engines. Maybe we can compromise in some way?

Reply:
I still consider these to be lower-level (from a game engine perspective) rendering configuration parameters. In an editor or general graphics configuration profile, they might map to Shadow Quality Low / Medium / High / Ultra.

This is the only place where it can be the lowest level of configuration, where we can select individual algorithms/methods and want to know their specific names so we know what implementation we are talking about. It is right now the only way to configure them at all, and I think we can advise on top of that.

I don't think we should try to hide the method names here. If we call them `Smooth` and `Stochastic` then there are different ways of implementing those kinds of results that can have different performance tradeoffs that developers will want to control because shadows may be more or less significant / challenging in the visuals of their game.

So now I think even `Stochastic` is too general and we should be more specific in naming it as the Call of Duty or Jimenez or so method.

That said, I do think the descriptions could be improved further.

Reply:
I edited the suggestion to both be more specific about the stochastic method, but also to be clearer about the quality and performance tradeoffs.

Reply:
Another option could be to reference them as if they were research papers, so `Castano13` and `Jimenez14`, though the CoD method uses Alan Wolfe's temporal noise modification from 2022, so maybe `Jimenez14Wolfe22`.

Reply:
Btw IGN is from the same CoD presentation. The blog post I linked just explains IGN better than the CoD slides. So, maybe remove Wolfe22.

Reply:
It's not exactly the same from what I could tell. The temporal aspect proposed there was:

> frameTick.x * IGN(uv) + frameTick.y
>
> where frameTick.x is 0.5 and frameTick.y is 0.0 for odd frames and 0.5 for even frames

Whereas Alan Wolfe's article says the temporal modification is:

>  scroll the texture by 5.588238 pixels each frame

But, they do attribute that to Jimenez in the article, so I'll remove Wolfe22.